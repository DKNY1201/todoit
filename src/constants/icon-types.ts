/*
To see the association between IconType and IconVersion, check App.tsx
 */
export enum IconType {
    Comment = 'comment-dots',
    DragDrop = 'grip-vertical',
    Schedule = 'calendar-alt',
    Action = 'ellipsis-h'
}

export enum IconVersion {
    Fas = 'fas', // Font awesome solid
    Far = 'far', // Font awesome regular
    Fal = 'fal', // Font awesome light
    Fab = 'fab' // Font awesome brands
}