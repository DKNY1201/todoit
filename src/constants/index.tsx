export const ADD_TODO = 'ADD_TODO';
export const EDIT_TODO = 'EDIT_TODO';
export const COMPLETE_TODO = 'COMPLETE_TODO';
export const DELETE_TODO = 'DELETE_TODO';

export type ADD_TODO = typeof ADD_TODO;
export type EDIT_TODO = typeof EDIT_TODO;
export type COMPLETE_TODO = typeof COMPLETE_TODO;
export type DELETE_TODO = typeof DELETE_TODO;