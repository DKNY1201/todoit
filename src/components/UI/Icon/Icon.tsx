import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as React from 'react';
import {IconType} from "../../../constants";

interface IIcon {
    clicked?: () => void;
    type: IconType;
}

const Icon = ({type, clicked}: IIcon) => {
    return (
        <span className={`icon icon-${type}`} onClick={clicked}>
            <FontAwesomeIcon icon={[type.version, type.name]} />
        </span>
    );
}

export default Icon;