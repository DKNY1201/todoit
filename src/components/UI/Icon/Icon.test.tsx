import * as enzyme from 'enzyme';
import * as React from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faCommentDots } from '@fortawesome/free-regular-svg-icons';

import {IconType, IconVersion} from '../../../constants';
import Icon from './Icon';

// Need to explicitly add icon here because test files is not under root component (App)
library.add(faCommentDots);

describe('<Icons > render', () => {
   it('should render a font awesome icon', () => {
      const wrapper = enzyme.mount(<Icon type={IconType.Comment} version={IconVersion.Far} />);
      expect(wrapper.find('.fa-' + IconType.Comment).length).toEqual(1);
   });
});