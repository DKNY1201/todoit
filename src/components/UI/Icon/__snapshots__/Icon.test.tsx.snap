// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icons > render should render a icon correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    type="comment-dots"
    version="far"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
                Array [
                        "far",
                        "comment-dots",
                      ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
/>,
      "className": "icon icon-far-comment-dots",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "border": false,
        "className": "",
        "fixedWidth": false,
        "flip": null,
        "icon": Array [
          "far",
          "comment-dots",
        ],
        "inverse": false,
        "listItem": false,
        "mask": null,
        "pull": null,
        "pulse": false,
        "rotation": null,
        "size": null,
        "spin": false,
        "symbol": false,
        "title": "",
        "transform": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
                    Array [
                              "far",
                              "comment-dots",
                            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
/>,
        "className": "icon icon-far-comment-dots",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "",
          "fixedWidth": false,
          "flip": null,
          "icon": Array [
            "far",
            "comment-dots",
          ],
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "title": "",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
